strapi/strapi
# Improve error when assets backup folder cannot be created

### What does it do?

- [x] It improves the error when the backup folder for the assets cannot be created
- [x] Avoids creating the backup folder if the assets are not going to be transferred. 

### Why is it needed?

To avoid unnecessary errors and if they are unavoidable, to have a better understanding on how to solve them.

### How to test it?
Create a strapi project inside a docker container with the public/uploads folder in a volume, so that there are permission issues and an error is thrown when trying to create the backup folder when transferring assets. 
Then, try to import and transfer data and you will see that the error is more specific than before.
And if you exclude the assets or specify only the content or the config, you won't see any error. 

```
# doocker-compose.yaml
version: '3'

services:
  strapi:
    build: .
    ports:
      - '1337:1337'
    volumes:
      - ./public/uploads:/opt/app/public/uploads
```
```
# Dockerfile
FROM node:16-alpine
# Installing libvips-dev for sharp Compatibility
RUN apk update && apk add --no-cache build-base gcc autoconf automake zlib-dev libpng-dev nasm bash vips-dev
ARG NODE_ENV=development
ENV NODE_ENV=${NODE_ENV}
WORKDIR /opt/
COPY ./package.json ./
ENV PATH /opt/node_modules/.bin:$PATH
RUN yarn config set network-timeout 600000 -g && yarn install
WORKDIR /opt/app
COPY ./ .
RUN yarn build
EXPOSE 1337
CMD ["yarn", "start"]
```

### Related issue(s)/PR(s)

https://github.com/strapi/strapi/issues/15613
